packages/components/src/spectrum/utils/useRenderNormalizedItem.tsx

```diff
@@ -50,7 +50,8 @@ export function useRenderNormalizedItem({
     (normalizedItem: NormalizedItem) => {
       const itemKey = getItemKey(normalizedItem);
       const content = wrapPrimitiveWithText(normalizedItem.item?.content);
-      const textValue = normalizedItem.item?.textValue ?? '';
+      const textValue =
```

Comment:
This should also be the same check as the first one. Something like:
```typescript
const textValue =
  normalizedItem.item?.textValue ??
  (itemKey == null ? undefined : String(itemKey));
```

Reply:
Namely the `as string` may not actually be true since it could be a `boolean` or a `number`. You need the `String` to ensure it is actually a string